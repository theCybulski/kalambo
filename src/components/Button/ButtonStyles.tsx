import styled, { css } from "styled-components";

export const Wrapper = styled.button`
  ${({ theme }) => css`
    position: relative;
    background-color: rgba(${theme.colors.primaryColor}, 1);
    color: #fff;
    border: none;
    border-radius: 10px;
    padding: 10px 15px;
    cursor: pointer;
    align-items: center;
  `};
`;
