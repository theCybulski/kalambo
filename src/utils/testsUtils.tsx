import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'mobx-react';
import { ThemeProvider } from 'styled-components';

import playerStore from '__mocks__/store/PlayerStore';
import roomStore from '__mocks__/store/RoomStore';
import rootStore from '__mocks__/store/RootStore';

import defaultTheme from 'shared/styles/defaultTheme';
import lightTheme from 'shared/styles/lightTheme';

export const testTheme = { ...defaultTheme, ...lightTheme };

export const mountWithTheme = (component) =>
  mount(<ThemeProvider theme={{ ...testTheme }}>{component}</ThemeProvider>);

export const mountWithMockedStore = (component) =>
  mount(
    <Provider {...{ playerStore, roomStore, rootStore }}>
      <ThemeProvider theme={{ ...testTheme }}>{component}</ThemeProvider>
    </Provider>
  );
