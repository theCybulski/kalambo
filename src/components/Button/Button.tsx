import React, { ButtonHTMLAttributes } from 'react';

import * as Styled from './ButtonStyles';

export interface IButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {}

const Button: React.FC<IButtonProps> = ({ children, ...props }) => (
  <Styled.Wrapper {...props}>{children}</Styled.Wrapper>
);

export default Button;
