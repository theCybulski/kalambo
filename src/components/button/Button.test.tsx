import React from 'react';
import toJson from 'enzyme-to-json';
import { mountWithTheme } from 'utils/tests/testsUtils';

import Button from './Button';

describe(`${Button.name}`, () => {
  it('Renders properly with any props', () => {
    const button = mountWithTheme(<Button type="submit">Test button</Button>);
    expect(toJson(button)).toMatchSnapshot();
  });

  it('Accepts various children', () => {
    const button = mountWithTheme(<Button>Test button</Button>);
    const button2 = mountWithTheme(
      <Button>
        <span>Another test button</span>
      </Button>
    );

    expect(button.text()).toBe('Test button');
    expect(button2.find('span')).toBeTruthy();
    expect(button2.find('span').text()).toBe('Another test button');
  });

  it('Calls onClick event', () => {
    const onClick = jest.fn();
    const button = mountWithTheme(<Button onClick={onClick}>Test button</Button>);

    button.simulate('click');

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
