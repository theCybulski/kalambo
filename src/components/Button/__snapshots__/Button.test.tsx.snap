// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Renders properly with any props 1`] = `
.c0 {
  position: relative;
  background-color: rgba(113,31,218,1);
  color: #fff;
  border: none;
  border-radius: 10px;
  padding: 10px 15px;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "additional": "50,50,50",
        "primaryColor": "113,31,218",
        "primaryFill": "255,255,255",
        "secondaryColor": "255,90,19",
        "secondaryFill": "249,248,253",
      },
      "deviceWidth": Object {
        "desktop": 1400,
        "desktopBig": 2000,
        "desktopSmall": 1260,
        "phone": 690,
        "phoneSmall": 370,
        "tablet": 1000,
      },
      "fonts": Object {
        "size": Object {
          "medium": "20px",
          "regular": "16px",
          "small": "12px",
        },
        "weight": Object {
          "bold": "700",
          "light": "300",
          "regular": "400",
        },
      },
      "shadows": Object {
        "primary": "0 10px 30px 0 rgba(84,37,201,0.06);",
      },
    }
  }
>
  <Button
    type="submit"
  >
    <styled.button
      type="submit"
    >
      <button
        className="c0"
        type="submit"
      >
        Test button
      </button>
    </styled.button>
  </Button>
</ThemeProvider>
`;
